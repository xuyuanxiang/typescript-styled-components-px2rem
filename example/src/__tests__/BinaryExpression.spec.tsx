import React from 'react';
import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import { BinaryExpression } from '../BinaryExpression';

const div: HTMLDivElement = document.createElement('div');

beforeEach(() => {
  document.body.appendChild(div);
});

afterEach(() => {
  ReactDOM.unmountComponentAtNode(div);
  document.body.removeChild(div);
});

it('should transform BinaryExpression', function() {
  TestUtils.act(() => {
    ReactDOM.render(<BinaryExpression id="btnId" />, div);
  });
  const button = div.querySelector('button#btnId');
  if (button) {
    expect(document.documentElement).toMatchSnapshot();
    const style = getComputedStyle(button);
    expect(style.width).toBe('2rem');
    expect(style.height).toBe('1rem');
    expect(style.padding).toBe('0.9rem 0.16rem 0.04rem 0.32rem');
    expect(style.lineHeight).toBe('0.18rem');
    expect(style.backgroundColor).toBe('red');
    expect(style.borderRadius).toBe('0.32rem');
  } else {
    throw new Error('BinaryExpression should be render');
  }
});
