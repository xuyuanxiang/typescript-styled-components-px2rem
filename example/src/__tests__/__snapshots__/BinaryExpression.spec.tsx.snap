// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform BinaryExpression 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.joKdKm{width:2rem;height:1rem;padding:0.9rem 0.16rem 0.04rem 0.32rem;line-height:0.18rem;border-radius:0.32rem;background-color:red;}
    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-bdVaJa joKdKm"
        id="btnId"
      />
    </div>
  </body>
</html>
`;
